### PR TITLE
Improve message for node or proxy join errors

### LINKED ISSUE #3003: Node failed to establish connection to cluster: x509:
**What happened**:
When testing revoking a token, I ended out this error message.  We should provide a clearer error message if the token has been revoked. 

```
ERRO [PROC:1]    Node failed to establish connection to cluster: x509: cannot validate certificate for 10.2.1.142 because it doesn't contain any IP SANs. service/connect.go:65
```

```
sudo teleport start \
>    --roles=node \
>    --token=078c4b7bd067a300a7a0ad197c656141 \
>    --ca-pin=sha256:00d0d05ad75c4afba64ca00851e6f4f6f162e2beeb6bc162aeff133a19c444cb \
>    --auth-server=10.2.1.142:3025
ERRO [PROC:1]    Node failed to establish connection to cluster: x509: cannot validate certificate for 10.2.1.142 because it doesn't contain any IP SANs. service/connect.go:65
ERRO [PROC:1]    Node failed to establish connection to cluster: x509: cannot validate certificate for 10.2.1.142 because it doesn't contain any IP SANs. service/connect.go:65
ERRO [PROC:1]    Node failed to establish connection to cluster: x509: cannot validate certificate for 10.2.1.142 because it doesn't contain any IP SANs. service/connect.go:65
ERRO [PROC:1]    Node failed to establish connection to cluster: x509: cannot validate certificate for 10.2.1.142 because it doesn't co
```

**What you expected to happen**:

We should provide a more helpful error message. e.g.
```
Node failed to establish connection to cluster:  Invalid Token. 
```

**Environment**:
- Teleport version (use `teleport version`): Teleport v4.1.0-beta.3 git:v4.1.0-beta.3-0-g28c51863 go1.12.1

### LINKED ISSUE #3316: Confusing error message with wrong ca_pin
## Description

In case if teleport proxy has a wrong `ca_pin` listed, the proxy join fails with message:

```
ERRO [PROC:1]    Proxy failed to establish connection to cluster: x509: certificate signed by unknown authority. time/sleep.go:149
```

This is very hard to debug and troubleshoot. Should detect a pin mismatch on join and give a proper message

### PR DESCRIPTION
**Pull request by awly**
The node first tries using the token with auth server. If that fails, ittries the same address as a proxy server.
If both fail, user only sees the error from the latter attempt. If using CA pin, this error will be `x509: certificate signed by unknown authority`, which is confusing.
    
Log both errors, and mention that a fallback is happening. The output looks like:
    
    ERRO [AUTH]      Failed to register through auth server: "my-hostname" [3e53a982-afd1-4d2f-8864-54c25fbe5865] can not join the cluster with role Node, the token is not valid; falling back to trying the proxy server auth/register.go:123
    ERRO [PROC:1]    Node failed to establish connection to cluster: failed to register through proxy server: x509: certificate signed by unknown authority. time/sleep.go:149

Fixes #3003
Fixes #3316

### CODE REVIEW

**Comment by awly** [comment]
This should also fix https://github.com/gravitational/teleport/issues/3316
Example output:
```
ERRO [AUTH]      Failed to register through auth server: provided certificate pin does not match cluster pin. This could have occurred if the Certificate Authority (CA) for the cluster was rotated, invalidating the old pin. Run "tctl status" to compare the pin used to join the cluster to the actual pin for the cluster.; falling back to trying the proxy server auth/register.go:126
ERRO [PROC:1]    Node failed to establish connection to cluster: failed to register through proxy server: x509: certificate signed by unknown authority. time/sleep.go:149
```

File: lib/service/service.go
Comment (webvictim) [inline lib/service/service.go:1598]: Are the changes to this file related to this PR? It looks like it might be mixed in from another change set.

**Review by benarent — APPROVED** [review]
I know these are small changes but these are huge help to improve our Ops/Dev Experience.

File: lib/service/service.go
Comment (awly) [reply lib/service/service.go:1598]: It is part of the PR.
When you spin up a node and it is stuck retrying to join the cluster, killing it (ctrl-c) causes a nil pointer panic. I noticed it while reproducing the issue and rolled up the fix.

Happy to split it into a separate PR if you think that's best.

Comment (webvictim) [reply lib/service/service.go:1598]: No, it’s fine, just seemed unrelated!

**Comment by awly** [comment]
@webvictim can I also get a codeowners approval?

**Review by webvictim — APPROVED** [review]

**Comment by awly** [comment]
retest this please

**Pull request merged**
